app/models/ability.rb

```diff
@@ -0,0 +1,32 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    #
+    return unless user.present?
+
+    can :read, :all
+    return unless user.admin?
+
+    can :manage, :all
```

Comment:
- [ ] The authorization is not well implemented. The post or comment author cannot really see the button to delete their post or comment (after modifying the view to use the `can?` method). Kindly fix that.

  ```
  return unless user.present?

  can :read, :all
  can :manage, Post, author: user
  can :manage, Comment, author: user

  return unless user.admin?

  can :manage, :all
  ```